packages/components/pie-toggle-switch/src/defs.ts

```diff
@@ -13,16 +15,24 @@ export interface ToggleSwitchProps extends RTLComponentProps {
     /**
      * Same as the HTML checked attribute - indicates whether the switch is on or off
      */
-    isChecked: boolean;
+    isChecked?: boolean;
     /**
      * Same as the HTML checked attribute - indicates whether the switch disabled or not
      */
-    isDisabled: boolean;
+    isDisabled?: boolean;
+    /**
+     * The label value of the component
+     */
+    label?: string;
+    /**
+     * The placement of the label such as leading or trailing
+     */
+    labelPlacement?: typeof labelPlacements[number];
 }
 
 /**
  * Event name for when the toggle switch checked state is changed.
  *
  * @constant
  */
-export const EVENT_TOGGLE_SWITCH_CHANGED = 'pie-toggle-switch-changed';
+export const ON_TOGGLE_SWITCH_CHANGED_EVENT = 'pie-toggle-switch-changed';
```

Comment:
More of a general TS question, shouldn't everything be typed?

Reply:
I mean, not only for the sake of declaring types, but also because TS might complain about where types might be lacking.

Reply:
typed as specifying that this variable should be a string? 

```
export const ON_TOGGLE_SWITCH_CHANGED_EVENT: string = 'pie-toggle-switch-changed';
```

if so, I'm not a big fan of redeclaring that the type should be string here as the compiler will infer that automatically as far as I know and we aren't reassigning it anywhere so I believe we don't have to.

Reply:
oh no, it should be automatically inferred by the compiler that's a string. 

if you try to reassign the variable to something else, it will complain.